### PR TITLE
Make Vector3d and Vector3 equal in interface

### LINKED ISSUE #678: Quaternion * Vector3 is not implemented for double precision
While there is a function for vector rotation with Quaternion and Vector3 there is none for Quaterniond and Vector3d

```C#
//exists
var newVec = new Quaternion() * new Vector3();

//doesn't exist
var newVec = new Quaterniond() * new Vector3d();
```

Version OpenTK 3.0pre

### PR DESCRIPTION
**Pull request by VPeruS**
Just copy-pasted code between classes. Fix #678

### CODE REVIEW

File: src/OpenTK/Math/Vector3.cs [hunk 1]
```diff
@@ -461,8 +460,7 @@ public static void ComponentMax(ref Vector3 a, ref Vector3 b, out Vector3 result
         }
 
         /// <summary>
-        /// Returns the Vector3 with the minimum magnitude. If the magnitudes are equal, the second vector
```
Comment (Nihlus) [inline src/OpenTK/Math/Vector3.cs]: Format edits to comments should not be included in this PR.

**Comment by Nihlus** [comment]
Please duplicate the unit tests for these methods as well.

**Comment by VPeruS** [comment]
You mean all test for "d" version or only add test for "f"?

File: src/OpenTK/Math/Vector3.cs [hunk 2]
```diff
@@ -1261,6 +1287,45 @@ public static Vector3 Unproject(Vector3 vector, float x, float y, float width, f
...
+        /// <param name="quat">The quaternion to rotate the vector by.</param>
+        /// <returns></returns>
+        public static Vector3 operator *(Quaternion quat, Vector3 vec)
+        {
+            Vector3 result;
+            Vector3.Transform(ref vec, ref quat, out result);
+            return result;
+        }
+
+        /// <summary>
```
Comment (VPeruS) [inline src/OpenTK/Math/Vector3.cs]: This was formatting as well

**Comment by Nihlus** [comment]
@VPeruS How's this PR coming along?

**Comment by VPeruS** [comment]
@Nihlus I completly forgot about this pr. Will continue on weekends

File: src/OpenTK/Math/Vector3d.cs [hunk 3]
```diff
@@ -1221,14 +1411,6 @@ public static void CalculateAngle(ref Vector3d first, ref Vector3d second, out d
             return new Vector3d(v3.X, v3.Y, v3.Z);
         }
 
-        /// <summary>Converts OpenTK.Vector3d to OpenTK.Vector3.</summary>
```
Comment (ThomasD13) [inline src/OpenTK/Math/Vector3d.cs]: Just a question: Why delete this method?

File: src/OpenTK/Math/Vector3.cs [hunk 4]
```diff
@@ -933,6 +928,56 @@ public static Vector3 Transform(Vector3 vec, Matrix3 mat)
...
+        public static Vector3 Transform(Vector3 vec, Matrix4 mat)
+        {
+            Vector3 result;
+            Transform(ref vec, ref mat, out result);
+            return result;
+        }
+
+        /// <summary>Transform a Vector by the given Matrix</summary>
+        /// <remarks>
+        /// It is incorrect to call this method passing the same variable for
```
Comment (ThomasD13) [inline src/OpenTK/Math/Vector3.cs]: Maybe use this PR to fix the build warning:
`Math/Vector3d.cs(932,62): warning CS1734: XMLcomment on 'Vector3d.Transform(ref Vector3d, ref Matrix4d, out Vector3d)' has a paramref tag for 'left', but there is no parameter by that name
        Math/Vector3d.cs(933,29): warning CS1734: XMLcomment on 'Vector3d.Transform(ref Vector3d, ref Matrix4d, out Vector3d)' has a paramref tag for 'right', but there is no parameter by that name`

File: src/OpenTK/Math/Vector3d.cs (hunk 3 above)
Comment (VPeruS) [reply src/OpenTK/Math/Vector3d.cs]: It was moved to OpenTK.Vector3

File: src/OpenTK/Math/Vector3.cs (hunk 4 above)
Comment (Nihlus) [reply src/OpenTK/Math/Vector3.cs]: Let's keep documentation changes separate from implementation changes.

**Comment by VPeruS** [comment]
Force pushed to remove all format changes

**Comment by Nihlus** [comment]
@VPeruS Status on the unit tests?

**Comment by Nihlus** [comment]
Thank you for you contribution! Currently, the `develop` branch is frozen while we work on the 4.0 release, and we are not accepting pull requests to it. All current pull requests should be based on that branch, or should be updated to be based on it.

In order to avoid clutter in our pull requests, we're going to be closing any pull request that isn't targeting 4.0 on the 31st of July, 2018. Please rebase your pull request before then.
